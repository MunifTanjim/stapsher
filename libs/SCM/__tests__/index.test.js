const helpers = require('../../../__tests__/helpers')

helpers.addSnapshotSerializers()

const { StapsherError } = require('../../Error')

const GitHub = require('../GitHub')

const SCM = require('../../SCM')

describe('libs/SCM', () => {
  let parameters = helpers.getParameters()

  it('returns instance of platform provider', () => {
    expect(SCM({ ...parameters, platform: 'github' })).toBeInstanceOf(GitHub)
  })

  it('throws error if platform not supported', () => {
    try {
      SCM({ ...parameters, platform: 'facebook' })
    } catch (err) {
      expect(err).toBeInstanceOf(StapsherError)
      expect(err).toMatchSnapshot()
    }
  })
})
