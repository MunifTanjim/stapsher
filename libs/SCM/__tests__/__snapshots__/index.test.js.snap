// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libs/SCM throws error if platform not supported 1`] = `
{
  "code": "UNSUPPORTED_SCM_PLATFORM",
  "statusCode": 400,
  "cause": {
    "platform": "facebook"
  }
}
`;
